src/QsCompiler/Transformations/BasicTransformations.cs

```diff
@@ -7,15 +7,33 @@
 using System.Linq;
 using Microsoft.Quantum.QsCompiler.DataTypes;
 using Microsoft.Quantum.QsCompiler.SyntaxTree;
+using Microsoft.Quantum.QsCompiler.Transformations.Core;
 
 
 namespace Microsoft.Quantum.QsCompiler.Transformations.BasicTransformations
 {
     // syntax tree transformations
 
     public class GetSourceFiles :
-        SyntaxTreeTransformation<NoScopeTransformations>
+        QsSyntaxTreeTransformation<GetSourceFiles.TransformationState>
     {
+        public class TransformationState
+        {
+            internal readonly HashSet<NonNullable<string>> SourceFiles = 
+                new HashSet<NonNullable<string>>();
+        }
+
+
+        private GetSourceFiles() :
+            base(new TransformationState())
+        { }
```

Comment:
nit: It might look cleaner to have this as a single line. Is there a style convention that we have that would mandate it look like this?

Reply:
I meant to go over the project and either put all colon on the old line or on the new line. Right now I think it's a wild mixture. The line can get pretty long if the call to the base constructor is on the same line, hence why I generally put it on a new line, but either one works for me.